### PR TITLE
fix: extract mod to file should respect path attribute

### PR DESCRIPTION
**Pull request by Young-Flash**
close https://github.com/rust-lang/rust-analyzer/issues/17181

### CODE REVIEW

File: crates/ide-assists/src/handlers/move_module_to_file.rs [hunk 1]
```diff
@@ -107,6 +116,72 @@ mod tests {
...
+fn test() {}
+mod $0expr {
+    struct A {}
+}
+"#,
+            r#"
+//- /parser/__mod.rs
+fn test() {}
+mod expr;
+//- /parser/expr.rs
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/move_module_to_file.rs:137]: This won't be correct, the module would need to be at `/parser/__mod/expr.rs` for this to be loaded no?

Comment (Young-Flash) [reply crates/ide-assists/src/handlers/move_module_to_file.rs:137]: `parser/expr.rs` can be loaded.

mod_with_path

IMO, with #path specified, `parser/__mod.rs` for mod `parser`(in the following code), can be seen as `parser/mod.rs`
```rust
#[path="parser/__mod.rs"]
mod parser;
```

Comment (Veykril) [reply crates/ide-assists/src/handlers/move_module_to_file.rs:137]: Right, thanks for double checking.

**Pull request merged**